File: ls215/regexpBook/regex.py
import re

'''
def p(text):
    print(re.findall(r'^c.t',
                     text,
                     flags=re.IGNORECASE))

p("cat")         # ['cat']
p("cot\n")       # ['cot']
p("CATASTROPHE") # ['CAT']
p("WILDCAUGHT")  # []
p("wildcat\n")   # []
p("-CET-")       # []
p("Yacht")       # []
'''

def p(regex, text):
    print(re.findall(regex,
                    text,
                    flags=re.IGNORECASE | re.MULTILINE))

text = ("cat\ncot\nCATASTROPHE\nWILDCAUGHT\n" +
        "wildcat\n-GET-\nYacht")

p(r'^c.t', text) # ['cat', 'cot', 'CAT']
p(r'c.t$', text) # ['cat', 'cot', 'cat', 'cht']

File: ls215/regexpBook/regex.js
function p(str, regex) {
  console.log(str.match(regex));
}

let text = "cat\ncot\nCATASTROPHE\nWILDCAUGHT\n" +
            "wildcat\n-GET-\nYacht"

p(text, /^c.t/mig);
p(text, /c.t$/mig);

/*
The match method returns a list of all matches that it finds.
For the first test, the matches were for "cat\n", "cot\n", and
"CATASTROPHE\n"; for the second test, "cat\n", "cot\n", "wildcat\n",
and "Yacht" all matched
*/